text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,944 @@
+# Meta
+[meta]: #meta
+- Name: Multi-arch support for builders and buildpack packages
+- Start Date: 2023-09-14
+- Author(s): @jjbustamante
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+The problem for adding support for multi-arch buildpacks can be divided into three parts:
+1. Support buildpack authors to **migrate their existing buildpacks** to support multi-arch.
+2. Support buildpack authors to **create new buildpacks and builders** that handle multi-arch from the beginning.
+3. Support application developers to **create application images** using multi-arch buildpacks and builders.
+
+The purpose of this RFC is to solve the statement 2, adding the capability to the commands:
+
+- `pack buildpack package`
+- `pack builder create`
+
+to create individuals OCI images artifacts per each os and arch (builders and buildpack packages) and handle the creation for the  [image index,](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+that combines them into one single consumable tag for end-users.
```

Comment:
To confirm, this will support `--publish` of the generated OCI images also? We'd been creating OCI archives in some of our pipelines and when we started generating multi-arch images in OCI archive files, it broke the tools we were using to move things around (`crane` and `skopeo`). It'd be really helpful to have that support in the tool.

Reply:
Yes, the idea is to push the generated OCI images and create the Image Index